### ISSUE TITLE
[Enhancement Request] Add setting to hide "reuselanguageinvoker" mismatch dialog

### ISSUE THREAD

**Issue by TwilightMercy** [Tikipeter/repository.tikipeter#282]
Hi Peter,

Until Kodi's bug regarding `reuselanguageinvoker` which crashes skins that uses widgets such as `FENtastic` / `Fenstruary` will be fixed, for `FENtastic` skin usage I turn `reuselanguageinvoker` to OFF.

In my build (using fully forked `Twilight/FEN 3.1.19` addon), I bring 2 options to choose to install: `Estuary` skin or `FENtastic` skin.

For `Estuary` skin build, Twilight addon deployed with addon.xml configured with:
 `<reuselanguageinvoker>true</reuselanguageinvoker>`

For `FENtastic` skin build, in order to avoid Kodi's bug crashes, `Twilight` addon deployed with `addon.xml` configured with: `<reuselanguageinvoker>false</reuselanguageinvoker>`

----------------------------------------------------------------------------------------------------------------------------------------------

However, I can only configure one instance of `Twilight` addon in my build's repository - [here](https://github.com/kodi7rd/repository/tree/master/zips/plugin.video.twilight)

So I chose to have in my official repo - the first option for `Estuary` skin, with `addon.xml` configured with:
 `<reuselanguageinvoker>true</reuselanguageinvoker>`

So, in that case it forces me to deploy in my `FENtastic` skin build - the addon with `reuselanguageinvoker` set to `false` in `addon.xml`, but installed as manual addon in Kodi, and not connected to any repository. and therefore, releasing versions updates in my repository only affects Estuary skin build, while `FENtastic` cannot be connected to repo since it will override `addon.xml` and will set it to `true`.

In order to solve this problem, I thought of connecting both builds to the same `Twilight` version from repository (with reuse **TRUE**),
but specific for FENtastic skin build - have it come with the configuration in:
**userdata\addon_data\plugin.video.twilight\settings.xml**

Reuse setting:
`<setting id="reuse_language_invoker">false</setting>`

While on first ever Kodi launch with the addon+settings.xml I described (addon.xml is set with TRUE, settings.xml is with to FALSE), - it will open the mismatch dialog:

![image](https://github.com/Tikipeter/repository.tikipeter/assets/29895680/873db5b0-5cfb-4e2c-9e35-cacc7b002d8b)

After clicking OK, it updates addon.xml to false and everything runs OK and no crashes, without having to restart Kodi again for changes to be 100% applied.

----------------------------------------------------------------------------------------------------------------------------------------------

I want to avoid the mismatch dialog, and just let it happen in the background, without having the user know/select OK/Cancel. Just to recognize that in `settings.xml` reuse is configured to FALSE, and update it aswell in the addon.

I tried to manually skip the dialog in the relevant code under:
**resources\lib\modules\service_functions.py**

**As you can here:**
![image](https://github.com/Tikipeter/repository.tikipeter/assets/29895680/f3f60115-ded3-40a1-bc3d-cffe6594bc6f)

After I first time launched Kodi - the code worked and updated `addon.xml` to false, but it froze Kodi for some seconds, and after that - the addon still crashed after exiting a movie, when returning to the home widgets menu. 

After the crash, I launched Kodi again and from there no issues - reuse lang is disabled and everything works good. It's only after the mismatch code (only because I skipped the mismatch dialog)

I couldn't understand why, since it's only a confirm dialog I skipped. but only with the dialog - there are no crashes after the mismatch code runs.

My request is if you can add a setting to FEN, to choose between showing the user the mismatch dialog or not - in order for it to affect immediately without the one-time crash in my code changes attempt)

Sorry for the long issue : ) 😅

**Comment by Tikipeter** [comment]
I'll add a setting to do this silently, but it won't fix any of the "freezing" issues you're having.

They are a result of the `update_local_addons` and `disable_enable_addon` code. Kodi will freeze when doing that.

**Comment by TwilightMercy** [comment]
Thank you very much for adding the setting. 
I’ll re-examine the behavior after it will be released

**Comment by TwilightMercy** [comment]
Hi, I've tested `FEN 3.1.20d` from test repository.

The setting itself works and I see that in `addon.xml` - the value silently changes to:
`<reuselanguageinvoker>false</reuselanguageinvoker>`

(in user settings configured `<setting id="reuse_language_invoker">false</setting>`)

When there is no mismatch to fix (i.e `addon.xml` + user `settings.xml` reuse lang is the same) - I see in kodi.log:
```
2023-07-27 09:07:41.090 T:17088    info <general>: ###FEN###: ReuseLanguageInvokerCheck Service Starting
2023-07-27 09:07:41.157 T:17088    info <general>: ###FEN###: ReuseLanguageInvokerCheck Service Finished
```

But:
When there is a mismatch (i.e `addon.xml` - true | user `settings.xml` reuse lang - `false`)  - I only see in the log:

`2023-07-27 08:57:35.257 T:16936    info <general>: ###FEN###: ReuseLanguageInvokerCheck Service Starting`

And that's it. The log is the same even if `Auto fix Invoker mismatch on Update` - is ON or OFF.

In addition, after the auto mismatch fix happens in the background, (`addon.xml` is updated from true to false due the user setting) - I still get one time Kodi crash (not immediately, after some navigating FEN menus / stopping movie playback and back to widgets home screen).

After the next Kodi restart when both settings are already synced, everything working and no crashes. (the one time crash only happens at mismatch fix Kodi run)

**Comment by TwilightMercy** [comment]
I did the following code change + test - commenting `execute_builtin('ActivateWindow(Home)', True)` line:

```
	if not get_setting('auto_invoker_fix') == 'true' and not confirm_dialog(text='%s\n%s' % (ls(33021), ls(33020))):
		return logger(twilight_str, 'ReuseLanguageInvokerCheck Service Finished')
	# COMMENTED - execute_builtin('ActivateWindow(Home)', True)
	update_local_addons()
	disable_enable_addon()
return logger(twilight_str, 'ReuseLanguageInvokerCheck Service Finished')
```

**It seems to work in this case:**

- In `addon.xml` - `<reuselanguageinvoker>true</reuselanguageinvoker>`
- In userdata `settings.xml` - `<setting id="reuse_language_invoker">false</setting>`
- In userdata `settings.xml` -  `<setting id="auto_invoker_fix">false</setting>`

The following happens:
1. The mismatch dialog opens

2. I press OK

3. `addon.xml` file is being updated to `<reuselanguageinvoker>false</reuselanguageinvoker>`

4. Kodi freezes for 2 seconds (when runs update_local_addons + disable_enable_addon functions)

5. Then immediately no crashes with widgets usage, and no Kodi restart is needed. everything works perfect from here.

And I see in log:
```
2023-08-05 13:40:05.413 T:4884     info <general>: ###TWILIGHT###: ReuseLanguageInvokerCheck Service Starting
2023-08-05 13:40:05.414 T:4884     info <general>: ###TWILIGHT###: ReuseLanguageInvokerCheck Service Finished
```

---------------------------------------------------------------------------------------------------------------------------------

However, in this case (with same code I shared, with the commented line):

- In `addon.xml` - `<reuselanguageinvoker>true</reuselanguageinvoker>`
- In userdata `settings.xml` - `<setting id="reuse_language_invoker">false</setting>`
- In userdata `settings.xml` -  `<setting id="auto_invoker_fix">true</setting>` **<--- CHANGE**

The following happens:
1. The mismatch dialog DOESN'T open

3. `addon.xml` file is being updated **IN THE BACKGROUND** to `<reuselanguageinvoker>false</reuselanguageinvoker>`

4. Kodi freezes forever, not only for 2 seconds. Then force closing it is needed.

5. After Kodi force Restart, `addon.xml` already comes with `<reuselanguageinvoker>false</reuselanguageinvoker>`, so no crashes from there.

I see in log:
```
2023-08-05 13:40:05.413 T:4884     info <general>: ###TWILIGHT###: ReuseLanguageInvokerCheck Service Starting
2023-08-05 13:40:05.414 T:4884     info <general>: ###TWILIGHT###: ReuseLanguageInvokerCheck Service Finished
```

But Kodi stays froze forever of course.

Thanks for being responsive for such a nit picky request!

**Comment by TwilightMercy** [comment]
@Tikipeter ? 🙏🏻

**Comment by TwilightMercy** [comment]
Hi Tikipeter, I don’t want to bother you that much since you’ve closed the issue, but I hope you’ve got the chance to read the results from my testing
